### ISSUE TITLE
Filled Bars e.g. Progress Bars

### ISSUE THREAD

**Issue by Snayff** [MyreMylar/pygame_gui#103]
**Is your feature request related to a problem? Please describe.**
Personally, I think the current healthbars are too proscriptive as they rely on the dev working ina  specific way. The rest of the UI Elements serve broad purposes with options to customise and as such can be tailored to fit the need.

**Describe the solution you'd like**
Similar to UIScreenSpaceHealthBar, but abstracted another layer. 

Simply put, a UI Element that we can size (as normal) and that takes a "% filled" that determines how much the outer, border rectangle shape is filled with the inner colour (as set by the theme). The inner fill could also be an image that is stretched or tiled.

**Comment by MyreMylar** [comment]
This may have snuck in to 0.6.0 as a hidden feature. 

I'm not sure when it was added exactly or if it works... Looks to be called `UIProgressBar`

**Comment by ConquerProgramming1** [comment]
Heh just posted a quasi proposal about this... Reference #244. Will look at UIProgressBar.
